projects/packages/connection/src/class-manager.php

```diff
@@ -842,10 +842,13 @@ public function xmlrpc_api_url() {
 	 * remain public because the call to action comes from the current site, not from
 	 * WordPress.com.
 	 *
+	 * @since 9.7.0 added $from param.
+	 *
 	 * @param String $api_endpoint (optional) an API endpoint to use, defaults to 'register'.
+	 * @param string $from An identifier to track where the registration is coming from. Can also be used to identify a site registred by a partner.
 	 * @return true|WP_Error The error object.
 	 */
-	public function register( $api_endpoint = 'register' ) {
+	public function register( $api_endpoint = 'register', $from = '' ) {
```

Comment:
yeah, I briefly thought about it but was unsure how and where add the filter, since a lot of redirects happen in the way. I'm gonna have a second look at it.

I did use it in the partners logic: https://github.com/Automattic/jetpack/pull/19609/files#diff-390fb19a16054ea62ccb8fef0b769e064b8599dff60e48e5aaf412eb5398aefeR56

Reply:
@zinigor what do you think of https://github.com/Automattic/jetpack/pull/19609/commits/95b8689c22ab6411de04f66431aff068d5b8eef8 ?

Reply:
We have discussed this a bit in Slack, and Leo came up with 3946a15, which seems to be good enough. Thanks!